python/ray/data/dataset.py

```diff
@@ -3400,6 +3400,11 @@ def _split(
                 left_metadata.append(ray.get(m0))
                 right_blocks.append(b1)
                 right_metadata.append(ray.get(m1))
+                # If return_right_half is requested, the input block b will be copied
+                # into b0 and b1. In such case, we can safely clear b if this is in
+                # lazy mode.
```

Comment:
Hmm isn't it copied in both cases?

Reply:
At the level of this method, the right half may not copied and conditioned on this boolean: https://sourcegraph.com/github.com/ray-project/ray@master/-/blob/python/ray/data/dataset.py?L3689

Reply:
But the left half is always copied. And the right half is either copied or None. It seems good to clear in either case.